lib/srv/authhandlers.go

```diff
@@ -716,3 +699,27 @@ func (h *AuthHandlers) authorityForCert(caType types.CertAuthType, key ssh.Publi
 func (h *AuthHandlers) isProxy() bool {
 	return h.c.Component == teleport.ComponentProxy
 }
+
+// MakeAuthAttemptFailureEvent creates an auth attempt failure event using
+// provided info.
+func MakeAuthAttemptFailureEvent(conn ssh.ConnMetadata, ident *sshca.Identity, err error) *apievents.AuthAttempt {
+	return &apievents.AuthAttempt{
+		Metadata: apievents.Metadata{
+			Type: events.AuthAttemptEvent,
+			Code: events.AuthAttemptFailureCode,
+		},
+		UserMetadata: apievents.UserMetadata{
+			Login:         conn.User(),
```

Comment:
Just a note this isn't one-to-one. `principal` got replaced with `conn.User()`. I assume this is ok.

Reply:
it's a good point. normally it is `conn.User` but sometimes you may want a different one (like in Git forwarder's case). i've did some refactor so the caller will jsut create the event themself. the event is not that big so this helper doesn't help that much anyway.